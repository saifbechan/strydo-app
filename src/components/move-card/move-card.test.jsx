import React from 'react';
import { useDrop } from 'react-dnd';
import { render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import mockStore from './../../redux/store.mock';

import MoveCard, { MoveCard as MoveCardPure } from './move-card.component';

jest.mock('react-dnd');

useDrop.mockImplementation(() => [{ isOver: false }, () => {}]);

const props = {
  idx: 0,
  columnId: 'abc-123'
};

it('renders correctly', () => {
  const { asFragment } = render(<MoveCardPure {...props} />);
  expect(asFragment()).toMatchSnapshot();
});

it('calls moveCard with the correct payload on click', () => {
  const store = mockStore();

  const { getByTestId } = render(<MoveCard store={store} {...props} />);

  fireEvent.drop(getByTestId('move-card-handler'));

  expect(store.getActions()).toEqual([]);
});
