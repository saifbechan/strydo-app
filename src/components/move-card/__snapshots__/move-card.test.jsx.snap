// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  height: 5px;
  padding: 5px;
}

.c1 {
  background: none;
  border: none;
  height: 100%;
}

<div
    class="c0"
    data-testid="move-card-handler"
  >
    <div
      class="c1"
    />
  </div>
</DocumentFragment>
`;
